plugins/callback/opentelemetry.py

```diff
@@ -350,7 +350,8 @@ def update_span_data(self, task_data, host_data, span, disable_logs, disable_att
         if not disable_logs:
             # This will avoid populating span attributes to the logs
             span.add_event(task_data.dump, attributes={} if disable_attributes_in_logs else attributes)
-            span.end(end_time=host_data.finish)
+        # Close span always
+        span.end(end_time=host_data.finish)
```

Comment:
Does it make sense to use a `try`/`finally` construct to ensure this is also closed in case of errors?

Reply:
I can even drop it:
- https://opentelemetry.io/docs/languages/python/instrumentation/#creating-nested-spans

>  # This span is also closed when it goes out of scope

I don't recall why I explicitly included it, but based on the existing docs, we can safely remove this.

Should I change the scope of this PR?

Reply:
I'm not sure what you mean.

Reply:
I misread how to create span and nested spans, there is no need to call `end` when using `with tracer.start_as_current_span("span-name") as span:`

The [opentelemetry.io/docs/languages/python/instrumentation/#creating-nested-spans](https://opentelemetry.io/docs/languages/python/instrumentation/#creating-nested-spans) says:

> When the 'with' block goes out of scope, 'span' is closed for you

so I'm gonna drop the `span.end` and update the changelog

Reply:
I think I spoke too fast... I cannot as I need to set the time when the span finished using the time when the ansible task finished.

Reply:
> Does it make sense to use a try/finally construct to ensure this is also closed in case of errors?

That will be handled by the `with tracer.start_as_current_span("span-name") as span:`

So we don't need it

Reply:
Ah ok. In that case everything sounds good :)